src/openforms/forms/migrations/0042_explanation_template_data_migration.py

```diff
@@ -0,0 +1,56 @@
+# Generated by Django 3.2.14 on 2022-08-26 15:29
+from django.db import migrations
+
+
+def append_explanation_template_message(apps, schema_editor):
+    from openforms.authentication.registry import register as auth_register
+    from django.utils.translation import gettext_lazy as _
+
+    Form = apps.get_model("forms", "Form")
+    for form in Form.objects.all():
+        plugins = form.authentication_backends
+        login_options_count = len(
+            [plugin_id for plugin_id in plugins if plugin_id in auth_register._registry]
```

Comment:
there's no need to use private API.

You can use either:

1. `auth_register.items()`
2. `plugin_id in auth_register`
3. `[plugin.identifier for plugin in auth_register]`

To get the currently available plugins.

Additionally, you can build this list of available plugins/plugin identifiers _outside_ of the for loop, as it doesn't change because of the loop.

Reply:
> Additionally, you can build this list of available plugins/plugin identifiers outside of the for loop, as it doesn't change because of the loop.

The`plugins` variable is coming from `form.authentication_backends`.

Reply:
I am talking about `auth_register`, and the registered plugins inside.

Reply:
I don't see a strong reason to make this change.

Reply:
`_registry` starts with an underscore prefix, indicating it is private API. That is plenty of reason to clean this up properly.

Reply:
@dariomory: Using a private API is bad practice, so please:

```suggestion
            [plugin_id for plugin_id in plugins if plugin_id in auth_register]
```